geomstats/geometry/complex_matrices.py

```diff
@@ -19,21 +19,23 @@ class ComplexMatrices(ComplexVectorSpace):
         Integers representing the shapes of the matrices: m x n.
     """
 
-    def __init__(self, m, n, **kwargs):
+    def __init__(self, m, n, equip=True):
         geomstats.errors.check_integer(n, "n")
         geomstats.errors.check_integer(m, "m")
-        kwargs.setdefault("metric", ComplexMatricesMetric(m, n))
-        kwargs.setdefault("default_point_type", "matrix")
-        super(ComplexMatrices, self).__init__(shape=(m, n), **kwargs)
+        super().__init__(shape=(m, n), equip=equip)
         self.m = m
         self.n = n
 
+    @staticmethod
+    def default_metric():
+        """Metric to equip the space with if equip is True."""
+        return ComplexMatricesMetric
```

Comment:
why not pass `m, n` here as well?

Reply:
The main idea is that `m` and `n` are available in the `space`. Therefore, when I pass the `space` to the `metric`, I don't need to pass anymore `m` and `n`. This way, we ensure the arguments of `space` and `metric` are always compatible without having to perform any check. We also ensure that `shape`, `default_point_type` and other (computational-oriented) attributes exist only in one place.

But, as you state above, this is the only "compatibility" check that is verified automatically. There's currently no way of prohibiting passing e.g. `Matrices` instead of `Hypersphere` to `HypersphereMetric`. It is also true for the converse, it is possible to wrongly do:

```python
space = Hypersphere(dim, equip=False)

space.equip_with_metric(MatricesMetric)
```

I think we probably don't want to restrict this. The main reason is that it may be hard to make a check that is restrictive enough, but not too restrictive: e.g. it would be easier to check `isinstance(metric, HypersphereMetric)`, but then we would not be able to equip the space with e.g. `ScalarProductMetric`. We would also restrict (power) user of developing their own independent metrics or spaces.